### ISSUE TITLE
`pushd` and `popd` should be treated like `cd`

### ISSUE THREAD

**Issue by LukeShu** [koalaman/shellcheck#863]
#### For bugs
- Rule Id (if any, e.g. SC1000): 
- My shellcheck version (`shellcheck --version` or "online"): 0.4.6
- [X] I tried on shellcheck.net and verified that this is still a problem on the latest commit
- [ ] It's not reproducible on shellcheck.net, but I think that's because it's an OS, configuration or encoding issue

#### For new checks and feature suggestions
- [X] shellcheck.net (i.e. the latest commit) currently gives no useful warnings about this
- [X] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```sh
#!/bin/bash
cd /foo
pushd /bar
popd
```

#### Here's what shellcheck currently says:

```
In demo.sh line 2:
cd /foo
^-- SC2164: Use 'cd ... || exit' or 'cd ... || return' in case cd fails.
```

#### Here's what I wanted or expected to see:

```
In demo.sh line 2:
cd /foo
^-- SC2164: Use 'cd ... || exit' or 'cd ... || return' in case cd fails.


In demo.sh line 3:
pushd /bar
^-- SC2164: Use 'pushd ... || exit' or 'pushd ... || return' in case pushd fails.


In demo.sh line 4:
popd
^-- SC2164: Use 'popd || exit' or 'popd || return' in case popd fails.
```
`pushd` and `popd` have exactly the same failure cases as `cd`; SC2164 should apply to them as well.

**Comment by arth1** [comment]
Exception for pushd -n, which always succeeds.

**Comment by mrshu** [comment]
The proposed changes in (#903) generate the output @LukeShu specified above, while also taking into account the `-n` option of `pushd` as mentioned by @arth1.

Hopefully, this issue can get closed soon.
